server.js

```diff
@@ -162,8 +166,34 @@ app.post("/login", async (req, res) => {
   }
 });
 
-// Authenticate user
 
+// User profile endpoint
+app.get('/users/:username', async (req, res) => {
```

Comment:
Need to sync as frontend route is 
<Route path="/:username" element={<UserProfile />} />